### PR TITLE
Add recipe for Annotation Cache Bundle

### PR DESCRIPTION
**Pull request by danieltoader**
| Q             | A
| ------------- | ---
| License       | MIT

### CODE REVIEW

**Comment by danieltoader** [comment]
@Nyholm Hello, I don't know what the procedure is to get this PR merged, could you please help or point me in the right direction?
Thank you!

File: emag-tech-labs/annotation-cache-bundle/5.0/config/packages/emag_annotation_cache.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+annotation_cache:
+    provider: {}
```
Comment (Nyholm) [inline emag-tech-labs/annotation-cache-bundle/5.0/config/packages/emag_annotation_cache.yaml:2]: I see that this bundle does not work without a provider. Let's add `cache.app` here as a default.

File: emag-tech-labs/annotation-cache-bundle/5.0/config/packages/emag_annotation_cache.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+    ignore_namespaces: []
```
Comment (Nyholm) [inline emag-tech-labs/annotation-cache-bundle/5.0/config/packages/emag_annotation_cache.yaml:3]: We can remove this. Let's add this to the docs instead. 

Alternatively, add `App\Entity` here. They are already cached by doctrine.

**Review by Nyholm — CHANGES_REQUESTED** [review]
Thank you for the PR. 

Unrelated: Have you done a test with Blackfire to make sure this is actually an improvement? 
Also, I made a similar bundle a few years ago. See https://github.com/Happyr/annotation-warmer

File: emag-tech-labs/annotation-cache-bundle/5.0/config/packages/emag_annotation_cache.yaml (hunk 2 above)
Comment (danieltoader) [reply emag-tech-labs/annotation-cache-bundle/5.0/config/packages/emag_annotation_cache.yaml]: Added 'App\Entity'

File: emag-tech-labs/annotation-cache-bundle/5.0/config/packages/emag_annotation_cache.yaml (hunk 1 above)
Comment (danieltoader) [reply emag-tech-labs/annotation-cache-bundle/5.0/config/packages/emag_annotation_cache.yaml]: Added 'cache.app' as the default value

**Comment by danieltoader** [comment]
@Nyholm After reading your comment I realized that the description was not that accurate so I updated the readme.md with more details about the Bundle and how it can be used.
Basically this package can be used to add cache to any method with the `@Cache` annotation. You can use any implementation that uses the PSR-6 standard as a provider and it will store the response of the method there.

This is helpful because you do not need to write the classic: search in cache, if found return, if not found run function code, store the result in cache and then return the value for each method you need to use cache for. That will all be done by the bundle which generates a proxy class that extends the original class and overwrites the methods with the `@Cache` annotation and adds the logic from above.

**Comment by Nyholm** [comment]
@symfony-flex-server review please

**Review by Nyholm — APPROVED** [review]
Okey. That is a bit different. 

Thank you

**Pull request merged**